lib/xmlhttprequest.js

```diff
@@ -299,74 +316,29 @@ class XMLHttpRequest extends XMLHttpRequestEventTarget {
     if (this.readyState !== XMLHttpRequest.OPENED) {
       throw new Error(); // todo
     }
-    if (typeof this._properties.uri === 'string') {
-      if (/^file:\/\//.test(this._properties.uri)) {
-        const p = this._properties.uri.replace(/^file:\/\//, '');
-        fs.stat(p, (err, stats) => {
-          if (!err) {
-            const response = fs.createReadStream(p);
-            response.statusCode = 200;
-            response.headers = {
-              'content-length': stats.size + '',
-            };
-            _setDispatchProgressEvents.call(this, response);
-            _receiveResponse.call(this, response);
-          } else {
-            const response = new stream.PassThrough();
-            response.statusCode = 500;
-            response.headers = {};
-            response.end(err.stack);
-            _setDispatchProgressEvents.call(this, response);
-            _receiveResponse.call(this, response);
-          }
-        });
-      } else {
-        // const async = this._flag.synchronous;
-        const client = utils.createClient(this._properties, response => {
-          const contentEncoding = response.headers['content-encoding'];
-          if (/^gzip$/i.test(contentEncoding)) {
-            const unzip = zlib.createGunzip();
-            for (let i = 0; i < RESPONSE_KEYS.length; i++) {
-              const k = RESPONSE_KEYS[i];
-              unzip[k] = response[k];
-            }
-            response = response.pipe(unzip);
-          }
-
-          _setDispatchProgressEvents.call(this, response);
-          _receiveResponse.call(this, response);
-        });
-        _properties.client = client;
-        Object.keys(_properties.requestHeaders).forEach(function(key) {
-          var value = _properties.requestHeaders[key];
-          client.setHeader(key, value);
-        });
-        _readyStateChange.call(this, XMLHttpRequest.HEADERS_RECEIVED);
-        if (typeof body === 'string') {
-          client.on('socket', _setDispatchProgressEvents.bind(this.upload));
-          client.end(body);
-        } else if (body instanceof FormData) {
-          client.on('socket', _setDispatchProgressEvents.bind(this.upload));
-          body.pipe(client);
-        } else {
-          client.end();
-        }
-        client.on('error', err => {
-          const errorEvent = new Event('error');
-          errorEvent.error = err;
-          this.dispatchEvent(errorEvent);
-        });
+    const {uri, method, headers} = _properties;
+
+      // Convert body to something fetch can consume
+    const toSendBody = body;
+    if (body instanceof FormData) {
+      toSendBody = {};
```

Comment:
I don't think this code is legal, per the above `const`.

Reply:
Good catch!